### PR TITLE
Bug 1316698 - EC2 metadata compat

### PR DESCRIPTION
**Pull request by jonasfj**
https://bugzilla.mozilla.org/show_bug.cgi?id=1316698

### CODE REVIEW

File: src/aws.js [hunk 1]
```diff
@@ -163,12 +174,26 @@ api.declare({
     DurationSeconds:    60 * 60   // Expire credentials in an hour
   }).promise();
 
+  // Make result compatibility with how EC2 metadata service let's instances
+  // access IAM roles
+  if (req.query.format === 'iam-role-compat') {
+    return res.status(200).json({
```
Comment (djmitche) [inline src/aws.js]: This avoids the schema validation.. clever :)

File: test/s3_test.js [hunk 2]
```diff
@@ -150,4 +150,28 @@ suite('aws S3 (STS)', () => {
...
+
+    let s3 = new aws.S3({
+      accessKeyId:     result.AccessKeyId,
+      secretAccessKey: result.SecretAccessKey,
+      sessionToken:    result.Token,
+    });
+    debug("### s3.putObject");
+    await s3.putObject({
+      Bucket:   bucket,
+      Key:      'folder1/' + id,
```
Comment (djmitche) [inline test/s3_test.js]: why create a new object on every test run?

**Comment by djmitche** [comment]
```
     Error: getaddrinfo ENOTFOUND taskclusterdev.table.core.windows.net taskclusterdev.table.core.windows.net:443
```
why does DNS suck?

**Comment by jonasfj** [comment]
> why does DNS suck?

Not a DNS issue `taskclusterdev` was my personal account after you leaked creds I just deleted it... So rotating tc-auth tests to use jungle account.

File: test/s3_test.js (hunk 2 above)
Comment (jonasfj) [inline test/s3_test.js]: to avoid consistency issues... this uses the garbage bucket where everything expires after 24 hours.

**Pull request merged**